src/main/java/frc/robot/commands/SnapToVisionTargetCommand.java

```diff
@@ -23,12 +22,13 @@
     private PIDController controller;
     private double offset;
     private double output = 0.0;
-    private RobotPose rp = RobotConstants.ROBOT_DEFAULTS.START_POSE;
-    
+    private PoseSource poseSource;
+    public static final double TIMEOUT_SECONDS=2;
 
-    public SnapToVisionTargetCommand(DriveSubsystem drive) {
-        super(drive);
+    public SnapToVisionTargetCommand(DriveSubsystem drive, PoseSource poseSource) {
```

Comment:
So we are using posesource as how the commands and subsystems get the pose instead of custom periodic...

Reply:
cool!